### PR TITLE
Addressed Transformer Rated Capacity of 0 hard crashes

### PR DESCRIPTION
**Pull request by Nigusse**
addressed  #6814

Added protection to avoid division by zero rated capacity in transformer model. This fix throws warning error message when the rated capacity is zero and the Performance Input Method is RATEDLOSSES. When the rated capacity is zero the transformer load and no load losses are set to zero and the simulation continues.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)

### CODE REVIEW

**Comment by Nigusse** [comment]
CI machine EnergyPlusFixture Test Summary shows PackagedTerminalHP_VSCoils_Sizing unit test failed. The failed unit test is not related to this fix. This branch is ready for review.

**Comment by nrel-bot-3** [comment]
@Nigusse @lgentile it has been 14 days since this pull request was last updated.

File: src/EnergyPlus/ElectricPowerServiceManager.cc [hunk 1]
```diff
@@ -3893,7 +3893,15 @@ ElectricTransformer::ElectricTransformer(std::string const &objectName)
             ShowContinueError("Invalid " + DataIPShortCuts::cAlphaFieldNames(6) + " = " + DataIPShortCuts::cAlphaArgs(6));
             errorsFound = true;
         }
-
+        if (ratedCapacity_ == 0) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/ElectricPowerServiceManager.cc:3904]: :+1: OK for one time check during get input.

File: src/EnergyPlus/ElectricPowerServiceManager.cc [hunk 2]
```diff
@@ -4104,7 +4112,7 @@ void ElectricTransformer::manageTransformers(Real64 const surplusPowerOutFromLoa
     } // switch usage mode
 
     // check availability schedule
-    if (ScheduleManager::GetCurrentScheduleValue(availSchedPtr_) > 0.0) {
+    if (ratedCapacity_ > 0.0 && ScheduleManager::GetCurrentScheduleValue(availSchedPtr_) > 0.0) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/ElectricPowerServiceManager.cc:4123]: Makes sense.

File: tst/EnergyPlus/unit/ElectricPowerServiceManager.unit.cc [hunk 3]
```diff
@@ -703,3 +704,106 @@ TEST_F(EnergyPlusFixture, ManageElectricPowerTest_CheckOutputReporting)
...
+    ScheduleManager::ScheduleInputProcessed = true;
+    DataEnvironment::DayOfYear_Schedule = General::JulianDay(DataEnvironment::Month, DataEnvironment::DayOfMonth, 1);
+    ScheduleManager::UpdateScheduleValues();
+
+    createFacilityElectricPowerServiceObject();
+    facilityElectricServiceObj->elecLoadCenterObjs.emplace_back(new ElectPowerLoadCenter(1));
+    facilityElectricServiceObj->elecLoadCenterObjs[0]->transformerObj = std::unique_ptr<ElectricTransformer>(new ElectricTransformer("TRANSFORMER"));
+    Real64 expectedtransformerObjLossRate = facilityElectricServiceObj->elecLoadCenterObjs[0]->transformerObj->getLossRateForOutputPower(2000.0);
+    // check the transformer loss rate for load and no load condition
+    EXPECT_EQ(expectedtransformerObjLossRate, 0.0);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ElectricPowerServiceManager.unit.cc:808]: A succinct, really nice unit test! :+1: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
I'm going to pull develop into this branch just because there have been some big changes there at IO freeze.  I don't want to accidentally let in a failing test due to an out of date IDF snippet or something.  I anticipate it being all green, but let's make sure.  If it's green in the morning, I'll merge this.

**Pull request merged**